src/printer.js

```diff
@@ -562,7 +562,20 @@ function genericPrintNoParens(path, options, print) {
       parts.push("return");
 
       if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+        if (n.argument.comments) {
+          parts.push(
+            n.loc.start.line === n.loc.end.line
```

Comment:
I'm afraid that this will not work. This is looking in the input where someone could have one giant line, instead what we want is to know if the output is going to be one line or not.

You can use ifBreak and willBreak to know this.